portfolio/src/main/java/com/google/sps/servlets/CasesDataServlet.java

```diff
@@ -0,0 +1,70 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.gson.Gson;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Scanner;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Returns case data as a JSON array, e.g. [{"lat": 38.4404675, "lng": -122.7144313}] */
+@WebServlet("/report")
+public class CasesDataServlet extends HttpServlet {
+  private Collection<Report> reports;
+
+  @Override
+  public void init() {
+    reports = new ArrayList<>();
+
+    Scanner scanner = new Scanner(getServletContext().getResourceAsStream("/WEB-INF/cases.csv"));
+    while (scanner.hasNextLine()) {
+      String line = scanner.nextLine();
+      String[] cells = line.split(",");
+
+      double lat = Double.parseDouble(cells[5]);
+      double lng = Double.parseDouble(cells[6]);
+      int active = Integer.parseInt(cells[10]);
+
+      reports.add(new Report(lat, lng, active));
+    }
+    scanner.close();
+  }
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    response.setContentType("application/json");
+    Gson gson = new Gson();
+    String json = gson.toJson(reports);
```

Comment:
It looks like "reports" never change once being initialized, and is only used here. If this is a finalized decision, you can define "json" as a member variable rather than "reports", so there is no need for this conversion for every get call.